### PR TITLE
CHEF-7691 Add resource updates for Client 18.4

### PR DESCRIPTION
**Pull request by IanMadd**
## Description

Add resource updates for Client 18.4

Related PRs:

- chef/chef#14052
- https://github.com/chef/chef/pull/13972
- https://github.com/chef/chef/pull/13394
- https://github.com/chef/chef/pull/14043

See #4156 for updates to apt_package.

## Issues Resolved

[List any existing issues this PR resolves, or any Discourse or
StackOverflow discussion that's relevant]

## Check List

- [ ] Spell Check
- [ ] Local build
- [ ] Examine the local build
- [ ] All tests pass

### CODE REVIEW

File: data/infra/resources/chocolatey_installer.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  description_list:
+  - markdown: If set, this overrides any configured proxy, proxy environment
+      variables, or parameters. This is enabled if set
+      to a value other than 'false' or '0'.
+- property: proxy_password
+  ruby_type: String
+  required: false
+  description_list:
+  - markdown: The password to use to build a proxy credential with. Will be consumed
+      by the proxy_credential property if both this property and proxy_user are set.
```
Comment (IanMadd) [inline data/infra/resources/chocolatey_installer.yaml]: I don't see a proxy_credential property.

File: data/infra/resources/chocolatey_installer.yaml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  ruby_type: String
+  required: false
+  description_list:
+  - markdown: Specifies the proxy URL to use during the download.
+- property: proxy_user
+  ruby_type: String
+  required: false
+  description_list:
+  - markdown: The username to use to build a proxy credential with. Will be consumed
+      by the proxy_credential property if both this property and proxy_password are
```
Comment (IanMadd) [inline data/infra/resources/chocolatey_installer.yaml]: Same here.

File: data/infra/resources/powershell_script.yaml [hunk 3]
```diff
@@ -128,6 +96,14 @@ properties_list:
   required: false
   description_list:
   - markdown: The current working directory from which the command will be run.
+- property: default_env
```
Comment (IanMadd) [inline data/infra/resources/powershell_script.yaml]: Is default_env valid here?

**Review by rahulgoel1 — APPROVED** [review]

File: data/infra/resources/chocolatey_installer.yaml (hunk 1 above)
Comment (johnmccrae) [reply data/infra/resources/chocolatey_installer.yaml]: "The user password used to connect to the proxy server with. If this is specified, the proxy user is also expected to be specified. "

File: data/infra/resources/chocolatey_installer.yaml (hunk 2 above)
Comment (johnmccrae) [reply data/infra/resources/chocolatey_installer.yaml]: "The user name used to connect to the proxy server with. If this is specified, the proxy password is also expected to be specified. "

File: data/infra/resources/powershell_script.yaml (hunk 3 above)
Comment (johnmccrae) [reply data/infra/resources/powershell_script.yaml]: This file appears to be pure garbage that got built by some errant AI. Some of the code it references comes from PR's merged in 2015. For now, you should nuke this file.

**Review by johnmccrae — APPROVED** [review]

**Pull request merged**